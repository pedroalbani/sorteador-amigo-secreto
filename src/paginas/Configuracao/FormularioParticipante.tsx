import { useRef, useState } from "react"
import { useAdicionarParticipante } from "../../state/hooks/useAdicionarParticipante"
import { useErro } from "../../state/hooks/useErro"

const FormularioParticipante = () => {

    const inputRef = useRef<HTMLInputElement>(null)
    const [novoParticipante, setNovoParticipante] = useState('')

    const erro = useErro()

    const adicionarParticipante = useAdicionarParticipante()

    const aoSubmeter = (evento: React.FormEvent<HTMLFormElement>) => {
        evento.preventDefault()
        adicionarParticipante(novoParticipante)
        setNovoParticipante('')
        inputRef.current?.focus()
    }

    return (<>
        <form onSubmit={aoSubmeter}>
            <div className="grupo-input-btn">
                <input
                    ref={inputRef}
                    value={novoParticipante}
                    onChange={evento =>  setNovoParticipante(evento.target.value)}
                    type="text"
                    required
                    placeholder='Insira o nome do participante' 
                    name="nome-do-participante"/>
                <button disabled={!novoParticipante} type='submit'>Adicionar</button>
            </div>
        </form>
        {erro && <p role="alert" className="alerta erro">{erro}</p>}
    </>)
}

export default FormularioParticipante