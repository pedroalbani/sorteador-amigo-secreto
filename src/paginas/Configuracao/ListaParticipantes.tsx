import { useListaParticipantes } from "../../state/hooks/useListaParticipantes"

const ListaParticipantes = () => {

    const participantes = useListaParticipantes()
   
    return (
        <ul>
            {participantes.map((nome, index) => <li key={index}>{nome}</li>)}
        </ul>
    )
}

export default ListaParticipantes