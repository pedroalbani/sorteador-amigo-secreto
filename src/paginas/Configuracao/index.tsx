import Card from "../../componentes/Card"
import FormularioParticipante from "./FormularioParticipante"
import ListaParticipantes from "./ListaParticipantes"
import Rodape from "./Rodape"

const Configuracao = () => {

    return (<Card>
        <section>
            <h2>Participantes</h2>
            <FormularioParticipante />
            <ListaParticipantes />
            <Rodape />
        </section>
    </Card>)
}

export default Configuracao