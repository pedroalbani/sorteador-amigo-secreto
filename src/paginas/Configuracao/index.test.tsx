import { render } from '@testing-library/react'
import { RecoilRoot } from 'recoil'
import Configuracao from '.'

jest.mock('react-router-dom', () => ({
    useNavigate: () => jest.fn(),
  }));

describe('a página de Configuracao', () => {
    test('impede o início da brincadeira', async () => {
        const { container } = render(
            <RecoilRoot>
                <Configuracao />
            </RecoilRoot>)
        expect(container).toMatchSnapshot()
    })
})