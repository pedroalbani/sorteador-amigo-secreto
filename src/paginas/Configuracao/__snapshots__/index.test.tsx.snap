// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`a página de Configuracao impede o início da brincadeira 1`] = `
<div>
  <div
    class="card"
  >
    <section>
      <h2>
        Participantes
      </h2>
      <form>
        <div
          class="grupo-input-btn"
        >
          <input
            name="nome-do-participante"
            placeholder="Insira o nome do participante"
            required=""
            type="text"
            value=""
          />
          <button
            disabled=""
            type="submit"
          >
            Adicionar
          </button>
        </div>
      </form>
      
      <ul />
      <footer
        class="rodape-configuracoes"
      >
        <button
          class="botao"
          disabled=""
        >
          Iniciar brincadeira!
        </button>
        <img
          alt="Sacolas de compras"
          src="/imagens/sacolas.png"
        />
      </footer>
    </section>
  </div>
</div>
`;
